### ISSUE TITLE
Could I get the gradient values of weights and biases?

### ISSUE THREAD

**Issue by ShiGuang72** [lululxvi/deepxde#149]

**Comment by ShiGuang72** [comment]
Hi Lu
Can I get the weight and bias gradient's values  through the DeepxDE package? I

**Comment by ShiGuang72** [comment]
To realize the dynamic allocation of the weight of the loss function, it is necessary to output the weight of the network and the partial derivative of the bias once, and realize the dynamic allocation of the weight of the loss function according to the weight and the derivative information of the bias, in which the loss function information includes the loss at the boundary and the loss at the internal point

**Comment by lululxvi** [comment]
You can get the values of weights and biases using `Model.state_dict()`[https://deepxde.readthedocs.io/en/latest/modules/deepxde.html#deepxde.model.Model.state_dict]. But if you want to get the `d Loss/dw`, you have to modify the code.

**Comment by ShiGuang72** [comment]
Thank you very much for answering my questions.
And I take the Model.state_dict() function. However, I can't get the gradient of weights and biases, And use the Model.state_dict() get the values of weights and biase.
And I have a another question: What the differences between Model.state_dict() and Model.print_model() I can get the values of weights and biases in this two ways

**Comment by ShiGuang72** [comment]
Thank you very much for answering my questions.
And I take the Model.state_dict() function. However, I can't get the gradient of weights and biases, And use the Model.state_dict() get the values of weights and biase.
And I have a another question: What the differences between Model.state_dict() and Model.print_model() I can get the values of weights and biases in this two ways.

**Comment by lululxvi** [comment]
The dict returned from state_dict() has the values of the optimizer, e.g., Adam. That might be the gradient, but I am not sure. Let me know if you know the answer.

They are almost the same, but Model.state_dict() returns more than Model.print_model(). Model.print_model() only has weights and biases, but Model.state_dict() has the variables associated to the optimizer.

**Comment by Rodfood** [comment]
I can get the variables of the weights and biases by using state_dict(), but I cann,t get the gradients of the weights and basises. because I want to correct the balance between the inter losses and the boundary  losses by using the information of the gradients of the weights .Can I do  it?

**Comment by ShiGuang72** [comment]
I can get the variables of the weights and biases by using state_dict(), but I cann,t get the gradients of the weights and basises. because I want to correct the balance between the inter losses and the boundary losses by using the information of the gradients of the weights .Can I do it?

**Comment by lululxvi** [comment]
I get what you mean. You might have to compute the gradients using `tf.gradients()`, see https://stackoverflow.com/questions/35226428/how-do-i-get-the-gradient-of-the-loss-at-a-tensorflow-variable . I didn't dig deep into this, and there might be other easier solutions.
